src/engine/source/hlp/src/SpecificParsers.cpp

```diff
@@ -376,3 +378,113 @@ bool parseURL(const char **it, char endToken, URLResult &result) {
+                                               "0123456789-_."};
+    static const int DOMAIN_MAX_SIZE = 253;
+    static const int LABEL_MAX_SIZE = 63;
+
+    const char *start = *it;
+    while (**it != endToken || **it != '\0') {
+        (*it)++;
+    }
+    std::string_view str { start, (size_t)((*it) - start) };
+
+    size_t protocol_end = str.find("://");
+    size_t domain_start = 0;
+    // Protocol
+    if (std::string::npos != protocol_end) {
+        result.protocol = str.substr(0, protocol_end);
+        domain_start = protocol_end + 3;
+    }
+    size_t domain_end = str.find("/", protocol_end + 3);
+    // Domain
+    result.domain = str.substr(domain_start, domain_end - domain_start);
+    // Domain length check
+    if (result.domain.empty() || result.domain.length() > DOMAIN_MAX_SIZE){
+        //TODO Log domain size error
+        return false;
+    }
+    // Domain valid characters check
+    for (char const &c: result.domain) {
+        if (std::string::npos == valid_chars.find(c)) {
+            //TODO Log invalid char error
+            return false;
+        }
+    }
+    // Route
+    if (std::string::npos != domain_end) {
+        result.route = str.substr(domain_end+1);
+    }
+
+    std::vector<std::string> labels;
+    size_t start_label = 0;
+    size_t end_label = 0;
+    while (end_label != std::string::npos) {
+        end_label = result.domain.find('.', start_label);
+        // TODO: Avoid String duplication here.
+        labels.emplace_back(result.domain.substr(start_label, end_label - start_label));
+        if (labels.back().empty() || labels.back().length() > LABEL_MAX_SIZE) {
+            //TODO Log label size error
+            return false;
+        }
+        start_label = end_label + 1;
+    }
+
+    // Guess the TLD
+    if (ccTLDlist.find(labels.back()) != ccTLDlist.end()) {
+        result.top_level_domain = labels.back();
+        labels.pop_back();
+    }
+    if (popularTLDlist.find(labels.back()) != popularTLDlist.end()) {
+        if (result.top_level_domain.empty()) {
+            result.top_level_domain = labels.back();
+        }
+        else {
+            result.top_level_domain = labels.back() + "." + result.top_level_domain;
+        }
+        labels.pop_back();
+    }
+
+    // Registered domain
+    if (result.top_level_domain.empty()) {
+        result.registered_domain = labels.back();
+    }
+    else {
+        result.registered_domain = labels.back() + "." + result.top_level_domain;
+    }
+    labels.pop_back();
+
+    // Subdomain
+    for (auto label : labels) {
+        if (result.subdomain.empty()) {
+            result.subdomain = label;
+        }
+        else {
+            result.subdomain = result.subdomain + "." + label;
+        }
+    }
+
+    // Address
+    result.address = result.domain;
+
+    // Validate if all fields are complete to identify a Fully Qualified Domain Name
+    if (!captureOpts.empty() && captureOpts[0] == "FQDN") {
```

Comment:
we probably want the "FQDN" to be a var with a name somewhere visible

Reply:
Changed!